src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,4 +156,12 @@ public function loadChoicesForValues(array $values, $value = null)
 
         return $this->loadChoiceList($value)->getChoicesForValues($values);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setChoiceFilter(callable $choiceFilter)
```

Comment:
Wild thoughts, as I don't know all the implications right now: I don't know if a new interface with a setter is a good idea. Shouldn't we simply inject in the loaders' constructors the `callable $choiceFilter` instead?

Reply:
:+1: makes sense, to keep compatibility with any choice_loader we could still decorate it if the choice filter wans injected initially.

Reply:
I expected this question :) My goal is enable this feature for all (more and more used) `CallbackChoiceLoader` in current projects, automatically.

Otherwise, the feature will be enabled only for core types (e.g. Intl form types, `EntityType`) and will require more code and then more maintenance (injecting the `$options['choice_filter']` everywhere).

This was the simplest solution I found to achieve a better DX, so callback loaders don't need to do something to enable this option.

Reply:
Also, this would allow for custom choice loader to enable this feature without care about injecting `$options['choice_filter']` everywhere the loader is used. We only care about the implementation as a plug-and-play interface.

Btw, there many examples of interface with a setter into the core, I'm not sure it's a  problem.